go/ros/ros2db3_to_mcap.go

```diff
@@ -71,6 +71,7 @@ func getSchemas(directories []string, types []string) (map[string][]byte, error)
 		}{
 			{parentPackage: rosPackage, rosType: rosType, schema: schema},
 		}
+		seenSubtypes := make(map[string]struct{})
```

Comment:
the magic of go...

do we need to start with an initial entry in this set? or maybe it would only matter if recursive types were allowed?

Reply:
> or maybe it would only matter if recursive types were allowed?

Are they allowed? I guess we would have run in infinite loops / max. call stack exceeded issues if this would be the case.

Reply:
I don't think they are allowed in ROS. I just wanted to think through if there are any other cases we should test to make sure this works

Reply:
https://github.com/ros2/ros2/issues/1445

Reply:
Then I better add it now, doesn't hurt